internal/cloudapi/v2/handler.go

```diff
@@ -503,6 +530,13 @@ func (h *apiHandlers) PostCompose(ctx echo.Context) error {
 			if request.Koji == nil {
 				return HTTPError(ErrorJSONUnMarshallingError)
 			}
+		} else if isLocalSave(ir.UploadOptions) {
+			// Override the image type upload selection and save it locally
+			// Final image is in /var/lib/osbuild-composer/artifacts/UUID/
+			irTarget = target.NewWorkerServerTarget()
+			irTarget.ImageName = imageType.Filename()
+			irTarget.OsbuildArtifact.ExportFilename = imageType.Filename()
+			irTarget.OsbuildArtifact.ExportName = imageType.Exports()[0]
```

Comment:
If I'm reading this right, in case one specified the `LocalUploadOptions` in the image request and the environment variable is not set, the `irTarget` will be `nil` and the image won't be uploaded anywhere. While in ideal case, the server should return an error saying that the specified upload target is not valid for the running configuration.

Reply:
Right, that's the same way it behaves now if the upload options are an empty object. The problem with the way it determines upload options is that it can't know that it is actually empty, the image type determines which struct it tries to marshal/unmarshal and it will end up being empty.

Reply:
Would it be possible to add an error in that case @thozza mentioned? It might save some nerves developers that forgot to set the right env variable (or made a typo).

Reply:
There already is, although I'm not sure if these errors make it back to the client. In the journal they look like:

```
Jun 15 11:35:06 fedora osbuild-worker[792]: time="2023-06-15T11:35:06-07:00" level=error msg="osbuild job failed: at least one target failed" jobId=37532cfb-aebf-45ea-8991-01f2d0af7d96
Jun 15 11:35:06 fedora osbuild-worker[792]: time="2023-06-15T11:35:06-07:00" level=error msg="failure details : [Code: 17, Reason: No AWS bucket provided, Details: org.osbuild.aws.s3]" jobId=37532cfb-aebf-45ea-8991-01f2d0af7d96
```

Reply:
These would make it to the client, but too late. Basically after waiting for the image to be built for couple of minutes, since it fails when the worker tries to upload the image.

In the case of `LocalUploadOptions`, the implementation should fail if the ENV variable is not set, without even enqueuing any job. You can specifically check if `local-save` is set to true, since the default value for `bool` type is `false`.

Reply:
Doh! good idea!